src/internal/packager/images/pull.go

```diff
@@ -426,15 +430,23 @@ func (i *ImageConfig) PullAll() (map[transform.Image]v1.Image, error) {
 	doneSaving <- 1
 	progressBarWaitGroup.Wait()
 
-	return refInfoToImage, nil
+	return imgInfoList, nil
 }
 
-// PullImage returns a v1.Image either by loading a local tarball or the wider internet.
-func (i *ImageConfig) PullImage(src string, spinner *message.Spinner) (img v1.Image, err error) {
+// PullImage returns a v1.Image either by loading a local tarball or pulling from the wider internet.
+func (i *ImageConfig) PullImage(src string, spinner *message.Spinner) (img v1.Image, hasImageLayers bool, err error) {
 	// Load image tarballs from the local filesystem.
 	if strings.HasSuffix(src, ".tar") || strings.HasSuffix(src, ".tar.gz") || strings.HasSuffix(src, ".tgz") {
 		spinner.Updatef("Reading image tarball: %s", src)
-		return crane.Load(src, config.GetCraneOptions(true, i.Architectures...)...)
+		img, err = crane.Load(src, config.GetCraneOptions(true, i.Architectures...)...)
+		if err != nil {
+			return nil, false, err
+		}
+		hasImageLayers, err = utils.HasImageLayers(img)
+		if err != nil {
+			return nil, false, fmt.Errorf("failed to check image %s layer mediatype: %w", src, err)
+		}
```

Comment:
I know that caching causes issues with these artifacts but there may be ways to simplify the logic to not have to state this check 3 times.

Reply:
I think I tackled this - mapping out the logic I believe the 3 separate spots were effectively an if/else-if/else so I wrapped them in that structure and moved to a single return statement for all success cases.